### ISSUE TITLE
[Bug] NVChad/Packer config issue: Installing Telekasten - how to pass local variable to config to set home dir

### ISSUE THREAD

**Issue by wakatara** [NvChad/NvChad#951]
**Describe the bug**

NVChad custom plugins under lua appear to not be able to have variables working in config (or I'm doing something very wrong) via a `local home = vim.fn.expand("~/custom_dir")` call preventing usage. Should this be set another way in the `config = function()` under NVchad (the lua works fine in vanilla neovim as a `lua << EOF ... EOF` directive.)?

Basically, I am unable to set a home directory in [telekasten](), the zettelekasten plugin. It works fine in vanilla neovim where I am using Plug. I have all other custom plugins working in NVChad under `/lua/custom/`. 

**To Reproduce**
I have the following block in my `lua/custom/plugins/init.lua` - all other plugins in the block work fine (and plugin in installed).

```lua
  {
    "renerocksai/telekasten.nvim",
    after = "telescope.nvim",
    config = function()
      -- this is the line that's problematic, it seems... 
      local home = vim.fn.expand("~/Documents/tk")
      require('telekasten').setup({
        home         = home,  -- have also tried just setting this to the directories directly and removing local home above
        take_over_my_home = true,
        auto_set_filetype = true,

        -- dir names for special notes (absolute path or subdir name)
        dailies      = home .. '/' .. 'dailies',
        weeklies     = home .. '/' .. 'weeklies',
        templates    = home .. '/' .. 'templates',

        -- image (sub)dir for pasting
        image_subdir = "img",

        -- markdown file extension
        extension    = ".md",

        -- following a link to a non-existing note will create it
        follow_creates_nonexisting = true,
        dailies_create_nonexisting = true,
        weeklies_create_nonexisting = true,

        -- template for new notes (new_note, follow_link)
        template_new_note = home .. '/' .. 'templates/new_note.md',
        template_new_daily = home .. '/' .. 'templates/daily.md',
        template_new_weekly= home .. '/' .. 'templates/weekly.md',

        -- image link style, wiki:v![[image name]], markdown: ![](image_subdir/xxxxx.png)
        image_link_style = "markdown",

        -- integrate with calendar-vim
        plug_into_calendar = true,
        calendar_opts = {
            -- calendar week display mode: 1 .. 'WK01', 2 .. 'WK 1', 3 .. 'KW01', 4 .. 'KW 1', 5 .. '1'
            weeknm = 2,
            -- use monday as first day of week: 1 .. true, 0 .. false
            calendar_monday = 4,
            -- calendar mark: where to put mark for marked days: 'left', 'right', 'left-fit'
            calendar_mark = 'left-fit',
        },
        -- telescope actions behavior
        close_after_yanking = false,
        insert_after_inserting = true,
        -- tag notation: '#tag', ':tag:', 'yaml-bare'
        tag_notation = "#tag",

        command_palette_theme = "dropdown",
        show_tags_theme = "dropdown",
        subdirs_in_links = false,
        template_handling = "smart",
        new_note_location = "smart",
        -- should all links be updated when a file is renamed
        rename_update_links = true,
      })
    end,
  }
```

**Expected behavior**

I would expect the telekasten plugin to pick up the home variable and incorporate it into the settings, so it can both find and create files for my zk system. Instead it keeps claiming no directory is configured/exists and offers to create `~/zettelkasten`. 

(I should also point out I've removed the `local home = vim.fn.expand("~/Documents/tk")` and simply replaced the `home = home` with ` home = "~/Documents/tk"` and `home = "/Users/me/Documents/tl"` but get the same error message. 

Basically, how *do* you get  variables inside a custom config block in NVChad?

**Screenshots**
nopes

**Desktop (please complete the following information):**
 - OSX 12.13.1
 - kitty 0.25 
 - neovim 0.6.1
 - all plugins PackerSync-ed

**Additional context**

Lua is a pretty new language to me, but excited by it being part of Neovim and trying to ramp up quickly as I have a plugin I want to build.

**Comment by wakatara** [comment]
@zbirenbaum 

I believe that comment only applies for Windows users (at least that's the way I read it), but will file a ticket with the author of telekasten to see if he has any insights. 

Thanks for looking at the issue, though!

**Comment by wakatara** [comment]
Heya @zbirenbaum

Thanks for taking the time to weigh in here a bit. The telelkasten folks are scratching their heads too, since the plugin works under standard Plug with neovim and the a `lua << EOF  { ... } EOF` mechanism so I think this is something particular to NVChad or, I now suspect, Packer. 

https://github.com/renerocksai/telekasten.nvim/issues/107

Let's see. I am sure whatever the config issue here is probably applicable to some other plugins people may be trying to wedge into NVChad, so let's see what we learn. Will keep you updated here. I'm sure it'll be interesting whatever it is... though it is quite odd that even hardcoded variables would not work in the `require.setup{ ... }`

It may be something with the internals of telekasten itself. I have to say I love the idea, so will lend a hand there to see if I can figure it out or if it is something about Lua or the dev just being on Linux that affects it. If it seems related to NVChad to Packer in NVChad, will bring it back here and re-open.

thanks again!

**Comment by wakatara** [comment]
@zbirenbaum 
(Oh, and totally understand the close out. Limited team. Time is a scarce resource. And imagine there are probably people who just try one thing, it does not work, and they just file a ticket, but I felt I'd done some pretty decent due diligence to see if it worked... =] ).

Also, once I get this working, would be happy to help out on issues in the future if you like. Happy to help out as my own code projects in Go have few issues at the moment and are pretty stable (and much less ambitious).

**Comment by wakatara** [comment]
@zbirenbaum 

OK, so figured it out. The issue is not with the variable at all but with the fact the packer directive need sto be `requires` rather than `after`.  8-/  Have now got it working fine in a slightly modified nvchad (though if we're dumping the custom directory I will have to look at perhaps the kickstarter template and some mannual config. 

I have to say I find the packer documentation on `requires`, `after`, and `wants` quite confusing, but there you go. 

Lemme know if you'd like me to make a PR to change the documentation for people that might be using NVChad and telekasten since I think it might fit in well with most powerusers as a plugin (or at least those that wanna use vim for everytghing as opposed to roam/logseq/org-roam/Notion. 

thanks for the help!
